test/test_simulation.py

```diff
@@ -1134,8 +1158,186 @@ def test_tag_filter(core: CoreSession, metrics):
+            submitter='user',
+            groups=['user'],
+            max_extracted=10000
+        ),
+        notification=dict(
+            queue='temp-data-monitor',
+            threshold=0
+        ),
+        files=[dict(
+            sha256=sha,
+            size=size,
+            name='abc123'
+        )]
+    )).as_primitives())
+
+    notification_queue = NamedQueue('nq-temp-data-monitor', core.redis)
+    dropped_task = notification_queue.pop(timeout=RESPONSE_TIMEOUT)
+    assert dropped_task
+    dropped_task = IngestTask(dropped_task)
+    sub: Submission = core.ds.submission.get(dropped_task.submission.sid)
+    assert len(sub.errors) == 0
+    assert len(sub.results) == 4, 'results'
+    assert core.pre_service.hits[sha] >= 2, f'pre_service.hits {core.pre_service.hits}'
+
+    # Wait until we get feedback from the metrics channel
+    metrics.expect('ingester', 'submissions_ingested', 1)
+    metrics.expect('ingester', 'submissions_completed', 1)
+    metrics.expect('dispatcher', 'submissions_completed', 1)
+    metrics.expect('dispatcher', 'files_completed', 1)
+
+    partial_results = 0
+    for res in sub.results:
+        result = core.ds.get_single_result(res, as_obj=True)
+        assert result is not None, res
+        if result.partial:
+            partial_results += 1
+    assert partial_results == 0, 'partial_results'
+
+
+def test_complex_extracted(core: CoreSession, metrics):
+    # stages to this processing when everything goes well
+    # 1. extract a file that will process to produce a partial result
+    # 2. hold a few seconds on the second stage of the root file to let child start
+    # 3. on the last stage of the root file produce the password
+    dispatcher.TIMEOUT_EXTRA_TIME = 10
+
+    child_sha, _ = ready_body(core, {
+        'pre': {'partial': {'passwords': 'test_temp_data_monitoring'}},
+    })
+
+    sha, size = ready_body(core, {
+        'pre': {
+            'response': {
+                'extracted': [{
+                    'name': child_sha,
+                    'sha256': child_sha,
+                    'description': 'abc',
+                    'classification': 'U'
+                }]
+            }
+        },
+        'core-a': {'lock': 5},
+        'finish': {'temporary_data': {'passwords': ['test_temp_data_monitoring']}},
+    })
+
+    core.ingest_queue.push(SubmissionInput(dict(
+        metadata={},
+        params=dict(
+            description="file abc123",
+            services=dict(selected=''),
+            submitter='user',
+            groups=['user'],
+            max_extracted=10000
+        ),
+        notification=dict(
+            queue='complex-extracted-file',
+            threshold=0
+        ),
+        files=[dict(
+            sha256=sha,
+            size=size,
+            name='abc123'
+        )]
+    )).as_primitives())
+
+    # Wait for the extract file to finish
+    metrics.expect('dispatcher', 'files_completed', 1)
```

Comment:
If I understand correctly, we could test for a `partial_results == 1` before calling `_global_semaphore.release()`?